src/libraries/System.Private.CoreLib/src/System/Threading/Wasi/WasiEventLoop.cs

```diff
@@ -58,13 +114,28 @@ internal static void DispatchWasiEventLoop()
 
             if (pending.Count > 0)
             {
+                var resolvedPollableIndex = -1;
+                // if there is CPU-bound work to do, we should not block on PollInterop.Poll below
+                // so we will append pollable resolved in 0ms
+                // in effect, the PollInterop.Poll would not block us
+                if (ThreadPool.PendingWorkItemCount > 0 || (s_mainTask != null && s_mainTask.IsCompleted))
+                {
+                    s_resolvedPollable ??= MonotonicClockInterop.SubscribeDuration(0);
```

Comment:
Should we check to see if `s_resolvedPollable` was already non-null here and either reuse it or dispose of it before reassigning it?